### PR TITLE
Updates on Frost Mage Guide (Orb, SP, Flurry, Prep)

### PR DESCRIPTION
**Pull request by earosselot**
### Description

Added Frozen Orb, Shifting Power, Flurry and Preparation to frost mage guide.

Some notes:
- I added a ShiftingPowerFrost class to analyze particular behaviors of this spell regarding frost specific things. Not sure if it is implemented well. Please take a look.
- To be able to generate different guide subsections for flurry and brain freeze analysis, I moved the analysis of Flurry casts to the Flurry module and imported into winters chill module. Again, I'm not sure if this is the correct way of doing it.

- Screenshot(s):
image
image
image

### CODE REVIEW

File: src/analysis/retail/mage/frost/core/BrainFreeze.tsx
Comment (emallson) [inline src/analysis/retail/mage/frost/core/BrainFreeze.tsx:61]: should this still live on `BrainFreeze` or ought it move to Flurry?

File: src/analysis/retail/mage/frost/talents/CometStorm.tsx
Comment (emallson) [inline src/analysis/retail/mage/frost/talents/CometStorm.tsx:192]: oops, good catch

File: src/analysis/retail/mage/frost/talents/Flurry.tsx
Comment (emallson) [inline src/analysis/retail/mage/frost/talents/Flurry.tsx]: typo
```suggestion
          At 2 or more {icicles} you are always able to shatter {glacialSpike}: <br />
```

File: src/analysis/retail/mage/frost/talents/Flurry.tsx
Comment (emallson) [inline src/analysis/retail/mage/frost/talents/Flurry.tsx]: would recommend trying `<p>` for these instead of `<br />`. the default spacing may not be what you want, but generally should prefer block/containing elements instead of `br`

```suggestion
        <p>{flurry} usage is important to make sure you can shatter as much {glacialSpike} as you can.
        You should only hold it after a {frostbolt} hardcast if you don't have {brainFreeze} and you
        have 0 or 1 {icicles}. </p>
        <p><small>
          At 2 or more {icicles} you are always able to shatter {glacialSpike}:
        </small></p>
```

File: src/analysis/retail/mage/frost/talents/FrozenOrb.tsx
Comment (emallson) [inline src/analysis/retail/mage/frost/talents/FrozenOrb.tsx]: same note about `<br/>`

**Review by emallson — COMMENTED** [review]
code looks fine. would recommend trying `<p>` instead of `<br/>`. otherwise will defer to @Sharrq on the magey bits of the changes

File: src/analysis/retail/mage/frost/core/BrainFreeze.tsx
Comment (earosselot) [reply src/analysis/retail/mage/frost/core/BrainFreeze.tsx:61]: yes, I think you're right. Maybe it belongs to the flurry module. I'll look into it.

Comment (earosselot) [reply src/analysis/retail/mage/frost/core/BrainFreeze.tsx:61]: I'm currently working on the brain freeze module, if it's ok with you I can do this change on the next PR, to avoid merge conflicts between this branch and the branch I working on now. Makes sense ?

Comment (emallson) [reply src/analysis/retail/mage/frost/core/BrainFreeze.tsx:61]: yea that is totally fine!

**Review by emallson — APPROVED** [review]
approve, pending sharrq's comments

**Comment by Sharrq** [comment]
I'll try and look today when I have time

**Comment by Sharrq** [comment]
I think this is fine. Not seeing anything that is an issue.

**Pull request merged**
